### DISCUSSION TITLE
RasterDataset with RBG Tif

### DISCUSSION THREAD

**Discussion by graceebc9** [torchgeo/torchgeo#598] [General]
Hello, I'm creating a new RasterDataset class for the NOAA high res RGB satellite imagery ([https://storms.ngs.noaa.gov/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fstorms.ngs.noaa.gov%2F&data=05%7C01%7Cgb669%40universityofcambridgecloud.onmicrosoft.com%7Cfeed6f321b7c420a98ba08da27623deb%7C49a50445bdfa4b79ade3547b4f3986e9%7C0%7C0%7C637865600139451254%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=a0kfqWjEJtFf0IgPYv8DPsMeKhT6rg2Y%2Fp5em2ibHSg%3D&reserved=0)) 

 but struggling to get the sampler to work, think it may be an issue with the TIF being multi-channel/ not sure how to extract the RGB bands from a single TIF thats not labelled. 

The data comes in TIF tiles of 2.5km, 3 x 9415 x 9415, and looks like this in XARRAY 
![image](https://user-images.githubusercontent.com/91670329/165335275-602de6c9-0791-4f73-b785-e47e3f50ff66.png)

This is my current class , for tiles with names like '20210830aC0932315w294800n.tif'
```python
class NOAA(RasterDataset):
    filename_glob = "*n.tif"
    filename_regex = "^(?P<date>\d{8})a.*$" 
    date_format = "%Y%m%d"
    
    seperate_files= False
    
    def __init__(
            self,
            root: str = "data",
            crs: Optional[CRS] = None,
            res: Optional[float] = None,
            # run: Sequence[str] = [],
            # bands: Optional[str] = [],
            transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
            cache: bool = True,
        ) -> None:
            """Initialize a new Dataset instance.

            """
            # self.bands = bands if bands else self.all_bands

            super().__init__(root, crs, res, transforms, cache,)
```

Thanks

**Comment by adamjstewart** [comment]
Could this be the same issue as https://github.com/microsoft/torchgeo/issues/466#issuecomment-1069383392?

**Comment by weiji14** [comment]
Could you point to a download link for the GeoTiff file (somehow can't get the download working on my browser)? Or maybe post the output of `gdalinfo 20210830aC0932315w294800n.tif`

My guess is that you'll need to read the RGB tif using rasterio's color interpretation, see https://rasterio.readthedocs.io/en/latest/topics/color.html. Note that xarray can't read the colorinterp properly, though I've been trying to get it to work via rioxarray (see https://github.com/corteva/rioxarray/pull/414).

**Comment by adamjstewart** [comment]
@graceebc9 does https://github.com/microsoft/torchgeo/issues/466#issuecomment-1069383392 solve your issue?

**Comment by adamjstewart** [comment]
Assuming that this issue has been resolved, but if not feel free to reopen this issue.
